### PR TITLE
fix(go-sdk): validation connection and model for batch operations

### PR DESCRIPTION
**Pull request by adriantam**
## Description
Add auth model validation/connection check on batch check/list relations/non transaction write.  This allows proper error to return  

## References
Related to  https://github.com/openfga/go-sdk/issues/21

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: config/clients/go/template/client/client.mustache [hunk 1]
```diff
@@ -416,6 +416,19 @@ func (client *{{appShortName}}Client) getAuthorizationModelId(authorizationModel
 	return modelId, nil
 }
 
+// helper function to validate the connection (i.e., get token)
+func (client *{{appShortName}}Client) validateAuthorizationModel(ctx _context.Context, authorizationModelId *string) error {
```
Comment (rhamzeh) [inline config/clients/go/template/client/client.mustache:420]: Can you call this `checkValidApiConnection`? The intention is to make sure we can talk to the API, not read the model.

Also to be consistent with the other SDKs

File: config/clients/go/template/client/client_test.mustache [hunk 2]
```diff
@@ -1367,7 +1474,7 @@ func Test{{appShortName}}Client(t *testing.T) {
 			t.Fatalf("%v", err)
 		}
 
-		if httpmock.GetTotalCallCount() != 4 {
+		if httpmock.GetTotalCallCount() != 5 {
 			t.Fatalf("{{appShortName}}Client.%v() - wanted %v calls to /check, got %v", test.Name, 4, httpmock.GetTotalCallCount())
```
Comment (rhamzeh) [inline config/clients/go/template/client/client_test.mustache:1478]: The error message here is now incorrect, it's 5 calls to check and 1 to check valid authorization model

Comment (adriantam) [reply config/clients/go/template/client/client_test.mustache]: fixed

File: config/clients/go/template/client/client.mustache (hunk 1 above)
Comment (adriantam) [reply config/clients/go/template/client/client.mustache]: updated

**Review by rhamzeh — APPROVED** [review]

**Pull request merged**